apps/system/utils/utils_cpuload.c

```diff
@@ -116,18 +117,21 @@ static void cpuload_stop(void)
  * So for zero load case, we check if avg_load is 0.0 or not.
  * Note: we compare with "-0.0" because if avg_load is 90.0, then
  * also we will get true if we compare with 0.0 */
+
 static bool has_cpuload(char *load_info)
 {
-#ifdef CONFIG_SMP
+#ifndef CONFIG_SMP
+	return (strcmp(load_info, "0.0") != 0);
+#else 
 	int len = strlen(load_info);
-	if (strcmp(&load_info[len - 4], "-0.0") == 0) {
-#else
-	if (strcmp(load_info, "0.0") == 0) {
-#endif
-		return false;
-	}
+	bool is_nonzero[CONFIG_SMP_NCPUS + 1] = {0, };
+	is_nonzero[0] = (strncmp(load_info, "0.0", 3) != 0);
+	is_nonzero[2] = (strncmp(&load_info[len - 4], "-0.0", 4) != 0);
+	load_info = strchr(load_info, '-'); 
+	is_nonzero[1] = (strncmp(load_info, "-0.0", 4) != 0);
```

Comment:
seems like we are only taking 2 core case with smp. Is it okay?

Reply:
Thank you for pointing it out. I fixed it.